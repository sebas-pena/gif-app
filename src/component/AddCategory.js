import React, { useState } from 'react'
import PropTypes from 'prop-types'



export const AddCategory = ({ setCategories, categories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault()
        if (inputValue && !categories.includes(inputValue)) {
            setCategories([inputValue, ...categories])
            setInputValue('')
        }
    }

    const handleInputChange = (e) => setInputValue(e.target.value)


    return (
        <form onSubmit={handleSubmit} >
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                name='category'
                placeholder='Add Category'
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
