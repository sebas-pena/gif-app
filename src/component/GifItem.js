import React from 'react'
import PropTypes from 'prop-types'
export const GifItem = ({ url, title }) => {
    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}