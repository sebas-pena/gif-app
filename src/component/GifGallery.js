import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'
import PropTypes from 'prop-types'

export const GifGallery = ({ category }) => {

    const { data, isLoading } = useFetchGifs(category)

    return (
        <div className=' animate__animated animate__fadeIn' >
            <h3>{category}</h3>
            <div className='gif-gallery'>
                {isLoading && <p>Loading...</p>}
                {
                    data.map(gif => <GifItem key={gif.id} {...gif} />)
                }
            </div>
            <hr />
        </div>
    )
}


GifGallery.propTypes = {
    category: PropTypes.string.isRequired
}