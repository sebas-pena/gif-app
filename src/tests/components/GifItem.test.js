import { GifItem } from "../../component/GifItem"
import { shallow } from 'enzyme'

describe('Probando el componente GifItem', () => {

    const url = "https://localhost:3200/images.jpg"
    const title = "title"
    let wrapper = shallow(<GifItem title={title} url={url} />)

    test('debe de renderizar correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('deben de coincidir los titulos', () => {
        let titleRendered = wrapper.find('p').text()
        expect(titleRendered).toBe(title)
    })


    test('la imagen debe contener la url y el alt correctos', () => {
        const img = wrapper.find('img')

        expect(img.prop("alt")).toBe(title)
        expect(img.prop("src")).toBe(url)

    })

    test('el contenedor debe tener las clases correctas', () => {

        const div = wrapper.find('div')

        expect(div.prop('className')).toBe("card animate__animated animate__fadeIn")
    })


})