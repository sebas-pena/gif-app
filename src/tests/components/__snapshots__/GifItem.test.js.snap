// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente GifItem debe de renderizar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost:3200/images.jpg"
  />
  <p>
    title
  </p>
</div>
`;

exports[`Probando el componente GifItem deben de coincidir los titulos 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost:3200/images.jpg"
  />
  <p>
    title
  </p>
</div>
`;
